lib/active_resource/base.rb

```diff
@@ -1283,14 +1330,27 @@ def persisted?
       @persisted
     end
 
+    # TODO remove: indifferent_access causes duplicates values and loss of reference
+    def attributes
+      super.with_indifferent_access
+    end
+
+    # TODO remove: ActiveModel::AttributeSet is private API
+    def attributes=(value)
+      case value
+      when ActiveModel::AttributeSet then @attributes = value
+      else super
+      end
+    end
```

Comment:
The `ActiveModel::Attributes#attributes=` interface relies on [ActiveModel::AttributeAssignment#attributes=](https://edgeapi.rubyonrails.org/classes/ActiveModel/AttributeAssignment.html#method-i-attributes-3D) for mass assignment.

This method override is for the sake of the `ActiveResource::Base#dup` implementation:

https://github.com/rails/activeresource/blob/9c8a2ee30325ff0bf789a239cc25fa361ec4a887/lib/active_resource/base.rb#L1345-L1350

If `#dup` were implemented differently, the `case` statement wouldn't be necessary and the override could be removed to completely rely on the behavior provided by its `super`.